leveldb/storage/storage.go

```diff
@@ -177,3 +178,121 @@ type Storage interface {
 	// called after the storage has been closed.
 	Close() error
 }
+
+// MeteredStorage collects read and write statistics.
+type MeteredStorage interface {
+	// Reads returns the cumulative number of read bytes of the underlying storage.
+	Reads() uint64
+	// Writes returns the cumulative number of written bytes of the underlying storage.
+	Writes() uint64
+}
```

Comment:
If you introduced a custom type, I don't think you need the interface too. Just call your custom type `meteredStorage` instead of `storageWrap` (bad name, doesn't say anything).

Reply:
The interface is needed at the type conversion, `db.go:966`

Reply:
I think the interface is needed at the type conversion, `db.go:966` in order to keep the custom type private.